win32ss/user/ntuser/window.c

```diff
@@ -1040,8 +1041,15 @@ VOID FASTCALL IntLinkHwnd(PWND Wnd, HWND hWndPrev)
         }
 
         if (Wnd == WndInsertAfter)
-            ERR("IntLinkHwnd -- Trying to link window 0x%p to itself!!\n", Wnd);
-        IntLinkWindow(Wnd, WndInsertAfter);
+        {
+            ERR("%s -- Trying to link window 0x%p to itself!!\n", __FUNCTION__, Wnd);
+            ASSERT(WndInsertAfter != Wnd);
+            FIXME("IntUnlinkWindow() was already called. Continuing as is seems wrong!");
```

Comment:
Looks to me like it's in a sense a similar case to the one above "Are we called with an erroneous handle" :
there reasoning here being that the given `WndInsertAfter` is "wrong", too. Not as a value itself, but by its relation with `Wnd` .

Reply:
same here. The exclamation mark is unnecessary. The log is full of them anyway. The several loggings fighting for the longest dick in this regard is a terrible development in every codebase.

Reply:
@HBelusca, can this case even happen, after `IntUnlinkWindow(Wnd)`?

> "Are we called with an erroneous handle"

If yes, then would a better fix be to check that `UserGetWindowObject(hWndPrev) != Wnd` at the start of current function, instead?